org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLRename.java

```diff
@@ -169,11 +172,58 @@ else if(node.isElement()) {
 					}
 				}
 			}
+
+			if (XSDUtils.isXSComplexType(element)) {
```

Comment:
We must create an extension (participant) for rename (liek we have for definition(IDefinitionParticpant, completion, etc) and move this code in XSDRenameParticpant (see XSDDefinitionParticipant for instance) because each extension should not set their feature in the same class. If you need help to create the particpant API IRenameParticpant, please tell me, but you have a lot of example like IHoverParticpant, ICompletionParticipant.